library/pk.c

```diff
@@ -322,14 +322,7 @@ int mbedtls_pk_can_do_ext(const mbedtls_pk_context *ctx, psa_algorithm_t alg,
     }
 
     psa_algorithm_t key_alg = psa_get_key_algorithm(&attributes);
-    /* Key's enrollment is available only when MBEDTLS_PSA_CRYPTO_CLIENT is
```

Comment:
I removed the comment because it was no more relevant IMO: the entire function is now guarded by MBEDTLS_PSA_CRYPTO_CLIENT which, according to the original comment, it means that "this function relies on a crypto implementation which is Mbed TLS compliant and therefore has also enrollment functions".
But perhaps I'm missing something like -> MBEDTLS_PSA_CRYPTO_CLIENT: there is a crypto provider, but it might be not Mbed TLS compliant?

Reply:
> Unofficially, we've agreed to help TF-M with using MBEDTLS_PSA_CRYPTO_CLIENT with alternative implementations of the PSA Crypto specification.

Ok, thanks for the explanation. So what about guarding both calls (set/get) to enrollment functions with `#if defined(MBEDTLS_PSA_CRYPTO_C)`?
When doing our usual tests we'll have `CRYPTO_C` and therefore `CRYPTO_CLIENT` so basically nothing changes. On TFM side they will have `CRYPTO_CLIENT` and `!CRYPTO_C` so they won't need to deal with those functions.

Reply:
At this stage and given we're in a hurry I propose to keep calls to the get function as they are. In addition to the one here, there's one in `mbedtls_pk_sign_ext`.

Reply:
> action on me in `mbedtls_pk_get_psa_attributes`

Actually, currently, in `development`, the call to `psa_set_key_enrollment_algorithm` is guarded by `#if defined(MBEDTLS_PSA_CRYPTO_C)`, since that's the condition for `mbedtls_pk_get_psa_attributes`. So actually, the action here is: in this pull request, when generalizing `mbedtls_pk_get_psa_attributes` to `CLIENT`, please keep a `C` guard around the call to `psa_set_key_enrollment_algorithm`, with a comment to explain why.